libs/designer/src/lib/core/utils/parameters/dynamicdata.ts

```diff
@@ -291,12 +293,18 @@ export async function getFolderItems(
   connectionReference: ConnectionReference | undefined,
   idReplacements: Record<string, string>
 ): Promise<TreeDynamicValue[]> {
-  const { definition, filePickerInfo } = dependencyInfo;
-  if (isLegacyDynamicValuesTreeExtension(definition) && filePickerInfo) {
-    const { open, browse } = filePickerInfo;
-    const { connectorId } = operationInfo;
-    const connectionId = connectionReference?.connection.id as string;
-    const { operationId, parameters: referenceParameters } = selectedValue ? browse : open;
+  const { definition, filePickerInfo, parameter } = dependencyInfo;
+
+  if (!filePickerInfo) {
+    throw new UnsupportedException('File picker information does not exist.');
```

Comment:
Can change this to throw the same error as below "`Dynamic extension '${definition.type}' is not implemented yet or not supported`", or something else. I thought it's cleaner to read the code if we return early in this condition.

Reply:
During initialization this is always set this should not be undefined can we remove this check?

Reply:
Lint complains that this could be undefined here. I'll move the filePickerInfo check back to the existing if condition. Was just trying to take out redundant common code for the two cases below.

Reply:
Updated